src/cpp/session/modules/SessionGit.cpp

```diff
@@ -204,10 +204,52 @@ std::string gitBin()
 }
 #endif
 
+bool waitForIndexLock(const FilePath& workingDir)
+{
+   using namespace boost::posix_time;
+   
+   // count the number of retries (avoid getting stuck in wait loops when
+   // an index.lock file exists and is never cleaned up)
+   static int retryCount = 0;
```

Comment:
Thanks, I understand better now. I was imagining a scenario in which a competing index.lock from e.g. a Git GUI caused RStudio to use up all its retries, after which it would no longer be able to wait a few ms for the file to clear until restarted. That seems pretty pessimistic too, though. :-) I'm fine with what you have here now that I know the rationale.

Do you think it'd be reasonable to try to delete the file if we believe it to be stale? (Presumably if it's not actually stale we will fail due to the file being locked by an owning process)